src/liballoc/collections/vec_deque.rs

```diff
@@ -2043,6 +2043,182 @@ impl<T> VecDeque<T> {
         }
     }
 
+    /// Returns a slice containing all elements of this deque if the internal storage is currently contiguous.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(deque_make_contiguous)]
+    ///
+    /// use std::collections::VecDeque;
+    ///
+    /// let mut buf = VecDeque::with_capacity(15);
+    ///
+    /// buf.push_back(2);
+    /// buf.push_back(1);
+    /// buf.push_front(3);
+    ///
+    /// assert!(buf.get_contiguous().is_none());
+    /// buf.make_contiguous();
+    ///
+    /// assert_eq!(buf.get_contiguous(), Some(&[3, 2, 1] as &[_]));
+    /// ```
+    #[unstable(feature = "deque_make_contiguous", issue = "none")]
+    pub fn get_contiguous(&self) -> Option<&[T]> {
+        if self.is_contiguous() {
+            unsafe { Some(&self.buffer_as_slice()[self.tail..self.head]) }
+        } else {
+            None
+        }
+    }
+
+    /// Returns a mutable slice containing all elements of this deque if the internal storage is currently contiguous.
+    ///
+    /// If this is not the case, this method does not modify this deque and returns
+    /// `None`. If this is not desired, consider using [`make_contiguous`](#method.make_contiguous) instead.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(deque_make_contiguous)]
+    ///
+    /// use std::collections::VecDeque;
+    ///
+    /// let mut buf = VecDeque::with_capacity(15);
+    ///
+    /// buf.push_back(2);
+    /// buf.push_back(1);
+    /// buf.push_front(3);
+    ///
+    /// assert!(buf.get_contiguous().is_none());
+    /// buf.make_contiguous();
+    ///
+    /// buf.get_contiguous_mut().unwrap()[2] = 42;
+    ///
+    /// assert_eq!(buf.get_contiguous(), Some(&[3, 2, 42] as &[_]));
+    /// ```
+    #[unstable(feature = "deque_make_contiguous", issue = "none")]
+    pub fn get_contiguous_mut(&mut self) -> Option<&mut [T]> {
```

Comment:
I can't actually think of a reason why one would use this instead of `make_contiguous`, so I don't mind if we remove this.

Mainly added it for completeness

Reply:
I feel that this method is mostly redundant when we already have `as_slices` and `as_slices_mut`.

Reply:
Well it gives you a single slice, not multiple, which sometimes could be exactly what you need. But yeah then you can just use make_contiguous.

Reply:
I do think that the immutable one is useful as it replaces the following (decently complex) snippet. But you are right ofc, `get_contiguous` is not absolutely necessary

```rust
let contiguous = if let (slice, &[]) = deque.as_slices() {
    Some(slice)
} else {
    None
}
```

Reply:
I'd personally be in favor of adding just `make_contiguous` and adding a blurb to the as_slices* functions that says that the first slice will have the whole contents after a call to make_contiguous (technically not guaranteed right now I think).